### ISSUE TITLE
server not started - errors when vagrant up

### ISSUE THREAD

**Issue by tbarross** [scrapinghub/portia#298]
i get lots of errors during vagrant up or vagrant provision..
i am running on a windows 8.1 machine, i got virtual box and vagrant installed.

if i try to open the vm it ask for portia user name and password which i dont have... 
or trying to browse localhost:8000 doesnt yield any results

could you help me please?

some of the command line output:

==> default: Unpacking replacement python ...
==> default: Preparing to replace libapt-pkg4.12 0.8.16~exp12ubuntu10 (using ...
/libapt-pkg4.12_0.8.16~exp12ubuntu10.25_i386.deb) ...
==> default: Unpacking replacement libapt-pkg4.12 ...
==> default: Processing triggers for man-db ...
==> default: Setting up libapt-pkg4.12 (0.8.16~exp12ubuntu10.25) ...
==> default: Processing triggers for libc-bin ...
==> default: ldconfig deferred processing now taking place
==> default: (Reading database ...
==> default: 30018 files and directories currently installed.)
==> default: Preparing to replace ca-certificates 20111211 (using .../ca-certifi
cates_20141019ubuntu0.12.04.1_all.deb) ...
==> default: Unpacking replacement ca-certificates ...
==> default: Preparing to replace libcurl3-gnutls 7.22.0-3ubuntu4 (using .../lib
curl3-gnutls_7.22.0-3ubuntu4.14_i386.deb) ...
==> default: Unpacking replacement libcurl3-gnutls ...
==> default: Preparing to replace libxml2 2.7.8.dfsg-5.1ubuntu4 (using .../libxm
l2_2.7.8.dfsg-5.1ubuntu4.11_i386.deb) ...
==> default: Unpacking replacement libxml2 ...
==> default: Selecting previously unselected package mysql-common.
==> default: Unpacking mysql-common (from .../mysql-common_5.5.44-0ubuntu0.12.04
.1_all.deb) ...
==> default: Selecting previously unselected package libmysqlclient18.
==> default: Unpacking libmysqlclient18 (from .../libmysqlclient18_5.5.44-0ubunt
u0.12.04.1_i386.deb) ...
==> default: Selecting previously unselected package libxslt1.1.
==> default: Unpacking libxslt1.1 (from .../libxslt1.1_1.1.26-8ubuntu1.3_i386.de
b) ...
==> default: Preparing to replace apt-transport-https 0.8.16~exp12ubuntu10 (usin
g .../apt-transport-https_0.8.16~exp12ubuntu10.25_i386.deb) ...
==> default: Unpacking replacement apt-transport-https ...
==> default: Selecting previously unselected package libstdc++6-4.6-dev.
==> default: Unpacking libstdc++6-4.6-dev (from .../libstdc++6-4.6-dev_4.6.3-1ub
untu5_i386.deb) ...
==> default: Selecting previously unselected package g++-4.6.
==> default: Unpacking g++-4.6 (from .../g++-4.6_4.6.3-1ubuntu5_i386.deb) ...
==> default: Selecting previously unselected package g++.
==> default: Unpacking g++ (from .../g++_4%3a4.6.3-1ubuntu5_i386.deb) ...
==> default: Selecting previously unselected package make.
==> default: Unpacking make (from .../make_3.81-8.1ubuntu1.1_i386.deb) ...
==> default: Selecting previously unselected package libdpkg-perl.
==> default: Unpacking libdpkg-perl (from .../libdpkg-perl_1.16.1.2ubuntu7.6_all
.deb) ...
==> default: Selecting previously unselected package dpkg-dev.
==> default: Unpacking dpkg-dev (from .../dpkg-dev_1.16.1.2ubuntu7.6_all.deb) ..
.
==> default: Selecting previously unselected package build-essential.
==> default: Unpacking build-essential (from .../build-essential_11.5ubuntu2.1_i
386.deb) ...
==> default: Selecting previously unselected package fakeroot.
==> default: Unpacking fakeroot (from .../fakeroot_1.18.2-1_i386.deb) ...
==> default: Selecting previously unselected package libalgorithm-diff-perl.
==> default: Unpacking libalgorithm-diff-perl (from .../libalgorithm-diff-perl_1
.19.02-2_all.deb) ...
==> default: Selecting previously unselected package libalgorithm-diff-xs-perl.
==> default: Unpacking libalgorithm-diff-xs-perl (from .../libalgorithm-diff-xs-
perl_0.04-2build2_i386.deb) ...
==> default: Selecting previously unselected package libalgorithm-merge-perl.
==> default: Unpacking libalgorithm-merge-perl (from .../libalgorithm-merge-perl
_0.08-2_all.deb) ...
==> default: Preparing to replace libevent-2.0-5 2.0.16-stable-1 (using .../libe
vent-2.0-5_2.0.16-stable-1ubuntu0.1_i386.deb) ...
==> default: Unpacking replacement libevent-2.0-5 ...
==> default: Selecting previously unselected package libevent-core-2.0-5.
==> default: Unpacking libevent-core-2.0-5 (from .../libevent-core-2.0-5_2.0.16-
stable-1ubuntu0.1_i386.deb) ...
==> default: Selecting previously unselected package libevent-extra-2.0-5.
==> default: Unpacking libevent-extra-2.0-5 (from .../libevent-extra-2.0-5_2.0.1
6-stable-1ubuntu0.1_i386.deb) ...
==> default: Selecting previously unselected package libevent-pthreads-2.0-5.
==> default: Unpacking libevent-pthreads-2.0-5 (from .../libevent-pthreads-2.0-5
_2.0.16-stable-1ubuntu0.1_i386.deb) ...
==> default: Selecting previously unselected package libevent-openssl-2.0-5.
==> default: Unpacking libevent-openssl-2.0-5 (from .../libevent-openssl-2.0-5_2
.0.16-stable-1ubuntu0.1_i386.deb) ...
==> default: Selecting previously unselected package libevent-dev.
==> default: Unpacking libevent-dev (from .../libevent-dev_2.0.16-stable-1ubuntu
0.1_i386.deb) ...
==> default: Selecting previously unselected package libexpat1-dev.
==> default: Unpacking libexpat1-dev (from .../libexpat1-dev_2.0.1-7.2ubuntu1.2_
i386.deb) ...
==> default: Selecting previously unselected package libmysqlclient-dev.
==> default: Unpacking libmysqlclient-dev (from .../libmysqlclient-dev_5.5.44-0u
buntu0.12.04.1_i386.deb) ...
==> default: Selecting previously unselected package libpython2.7.
==> default: Unpacking libpython2.7 (from .../libpython2.7_2.7.3-0ubuntu3.8_i386
.deb) ...
==> default: Selecting previously unselected package libxml2-dev.
==> default: Unpacking libxml2-dev (from .../libxml2-dev_2.7.8.dfsg-5.1ubuntu4.1
1_i386.deb) ...
==> default: Selecting previously unselected package libxslt1-dev.
==> default: Unpacking libxslt1-dev (from .../libxslt1-dev_1.1.26-8ubuntu1.3_i38
6.deb) ...
==> default: Selecting previously unselected package python2.7-dev.
==> default: Unpacking python2.7-dev (from .../python2.7-dev_2.7.3-0ubuntu3.8_i3
86.deb) ...
==> default: Selecting previously unselected package python-dev.
==> default: Unpacking python-dev (from .../python-dev_2.7.3-0ubuntu2.2_i386.deb
) ...
==> default: Selecting previously unselected package unattended-upgrades.
==> default: Unpacking unattended-upgrades (from .../unattended-upgrades_0.76ubu
ntu1.1_all.deb) ...
==> default: Selecting previously unselected package python-pycurl.
==> default: Unpacking python-pycurl (from .../python-pycurl_7.19.0-4ubuntu3_i38
6.deb) ...
==> default: Selecting previously unselected package python-software-properties.

==> default: Unpacking python-software-properties (from .../python-software-prop
erties_0.82.7.7_all.deb) ...
==> default: Selecting previously unselected package libffi-dev.
==> default: Unpacking libffi-dev (from .../libffi-dev_3.0.11~rc1-5_i386.deb) ..
.
==> default: Processing triggers for man-db ...
==> default: Processing triggers for ureadahead ...
==> default: ureadahead will be reprofiled on next reboot
==> default: Processing triggers for install-info ...
==> default: Setting up libssl-dev (1.0.1-4ubuntu5.31) ...
==> default: Setting up libexpat1 (2.0.1-7.2ubuntu1.2) ...
==> default: Setting up python2.7 (2.7.3-0ubuntu3.8) ...
==> default: Setting up python (2.7.3-0ubuntu2.2) ...
==> default: Setting up ca-certificates (20141019ubuntu0.12.04.1) ...
==> default: Updating certificates in /etc/ssl/certs...
==> default: 38 added, 17 removed; done.
==> default: Running hooks in /etc/ca-certificates/update.d....
==> default: done.
==> default: Setting up libcurl3-gnutls (7.22.0-3ubuntu4.14) ...
==> default: Setting up libxml2 (2.7.8.dfsg-5.1ubuntu4.11) ...
==> default: Setting up mysql-common (5.5.44-0ubuntu0.12.04.1) ...
==> default: Setting up libmysqlclient18 (5.5.44-0ubuntu0.12.04.1) ...
==> default: Setting up libxslt1.1 (1.1.26-8ubuntu1.3) ...
==> default: Setting up apt-transport-https (0.8.16~exp12ubuntu10.25) ...
==> default: Setting up make (3.81-8.1ubuntu1.1) ...
==> default: Setting up libdpkg-perl (1.16.1.2ubuntu7.6) ...
==> default: Setting up dpkg-dev (1.16.1.2ubuntu7.6) ...
==> default: Setting up fakeroot (1.18.2-1) ...
==> default: update-alternatives:
==> default: using /usr/bin/fakeroot-sysv to provide /usr/bin/fakeroot (fakeroot
) in auto mode.
==> default: Setting up libalgorithm-diff-perl (1.19.02-2) ...
==> default: Setting up libalgorithm-diff-xs-perl (0.04-2build2) ...
==> default: Setting up libalgorithm-merge-perl (0.08-2) ...
==> default: Setting up libevent-2.0-5 (2.0.16-stable-1ubuntu0.1) ...
==> default: Setting up libevent-core-2.0-5 (2.0.16-stable-1ubuntu0.1) ...
==> default: Setting up libevent-extra-2.0-5 (2.0.16-stable-1ubuntu0.1) ...
==> default: Setting up libevent-pthreads-2.0-5 (2.0.16-stable-1ubuntu0.1) ...
==> default: Setting up libevent-openssl-2.0-5 (2.0.16-stable-1ubuntu0.1) ...
==> default: Setting up libevent-dev (2.0.16-stable-1ubuntu0.1) ...
==> default: Setting up libexpat1-dev (2.0.1-7.2ubuntu1.2) ...
==> default: Setting up libmysqlclient-dev (5.5.44-0ubuntu0.12.04.1) ...
==> default: Setting up libpython2.7 (2.7.3-0ubuntu3.8) ...
==> default: Setting up libxml2-dev (2.7.8.dfsg-5.1ubuntu4.11) ...
==> default: Setting up libxslt1-dev (1.1.26-8ubuntu1.3) ...
==> default: Setting up python2.7-dev (2.7.3-0ubuntu3.8) ...
==> default: Setting up python-dev (2.7.3-0ubuntu2.2) ...
==> default: Setting up unattended-upgrades (0.76ubuntu1.1) ...
==> default: Setting up python-pycurl (7.19.0-4ubuntu3) ...
==> default: Setting up python-software-properties (0.82.7.7) ...
==> default: Setting up libffi-dev (3.0.11~rc1-5) ...
==> default: Setting up libstdc++6-4.6-dev (4.6.3-1ubuntu5) ...
==> default: Setting up g++-4.6 (4.6.3-1ubuntu5) ...
==> default: Setting up g++ (4:4.6.3-1ubuntu5) ...
==> default: update-alternatives:
==> default: using /usr/bin/g++ to provide /usr/bin/c++ (c++) in auto mode.
==> default: Setting up build-essential (11.5ubuntu2.1) ...
==> default: Processing triggers for libc-bin ...
==> default: ldconfig deferred processing now taking place
==> default: Installing python pip
==> default: =====================
==> default: 2015-10-13 10:49:19 URL:https://bootstrap.pypa.io/get-pip.py [14127
44/1412744] -> "-" [1]
==> default: Collecting pip
==> default: /tmp/tmpjuMJiQ/pip.zip/pip/_vendor/requests/packages/urllib3/util/s
sl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. T
his prevents urllib3 from configuring SSL appropriately and may cause certain SS
L connections to fail. For more information, see https://urllib3.readthedocs.org
/en/latest/security.html#insecureplatformwarning.
==> default:   Downloading pip-7.1.2-py2.py3-none-any.whl (1.1MB)
==> default: Collecting setuptools
==> default:   Downloading setuptools-18.4-py2.py3-none-any.whl (462kB)
==> default: Collecting wheel
==> default:   Downloading wheel-0.26.0-py2.py3-none-any.whl (63kB)
==> default: Installing collected packages: pip, setuptools, wheel
==> default: Successfully installed pip-7.1.2 setuptools-18.4 wheel-0.26.0
==> default: /tmp/tmpjuMJiQ/pip.zip/pip/_vendor/requests/packages/urllib3/util/s
sl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. T
his prevents urllib3 from configuring SSL appropriately and may cause certain SS
L connections to fail. For more information, see https://urllib3.readthedocs.org
/en/latest/security.html#insecureplatformwarning.
==> default: Installing Splash
==> default: =================
==> default: gpg:
==> default: keyring `/tmp/tmpsSvNbb/secring.gpg' created
==> default: gpg:
==> default: keyring`/tmp/tmpsSvNbb/pubring.gpg' created
==> default: gpg:
==> default: requesting key 779C27D7 from hkp server keyserver.ubuntu.com
==> default: gpg:
==> default: /tmp/tmpsSvNbb/trustdb.gpg: trustdb created
==> default: gpg:
==> default: key 779C27D7: public key "Launchpad PPA for pi-rho" imported
==> default: gpg:
==> default: Total number processed: 1
==> default: gpg:
==> default:               imported: 1
==> default:   (RSA: 1)
==> default: OK
==> default: E
==> default: :
] is not known.mmand line option '
==> default: Reading package lists...
==> default: Building dependency tree...
==> default: Reading state information...
==> default: E
==> default: :
==> default: Unable to locate package
==> default: /vagrant/docker/splash.sh: line 5: ca-certificates: command not fou
nd
==> default: /vagrant/docker/splash.sh: line 6: xvfb: command not found
==> default: /vagrant/docker/splash.sh: line 7: pkg-config: command not found
==> default: /vagrant/docker/splash.sh: line 8: libqt4-webkit: command not found

==> default: /vagrant/docker/splash.sh: line 9: python-qt4: command not found
==> default: /vagrant/docker/splash.sh: line 10: libre2: command not found
==> default: /vagrant/docker/splash.sh: line 11: libicu48: command not found
==> default: /vagrant/docker/splash.sh: line 12: zlib1g: command not found
==> default: /vagrant/docker/splash.sh: line 13: liblua5.2: command not found
==> default: /vagrant/docker/splash.sh: line 14: $'\r': command not found
==> default: Exception:
==> default: Traceback (most recent call last):
==> default:   File "/usr/local/lib/python2.7/dist-packages/pip/basecommand.py",
 line 211, in main
==> default:     status = self.run(options, args)
==> default:   File "/usr/local/lib/python2.7/dist-packages/pip/commands/install
.py", line 282, in run
==> default:     wheel_cache
==> default:   File "/usr/local/lib/python2.7/dist-packages/pip/basecommand.py",
 line 272, in populate_requirement_set
==> default:     wheel_cache=wheel_cache
==> default:   File "/usr/local/lib/python2.7/dist-packages/pip/req/req_install.
py", line 213, in from_line
==> default:     wheel_cache=wheel_cache, constraint=constraint)
==> default:   File "/usr/local/lib/python2.7/dist-packages/pip/req/req_install.
py", line 67, in **init**
==> default:     req = pkg_resources.Requirement.parse(req)
==> default:   File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_reso
urces/__init__.py", line 2985, in parse
==> default:     raise ValueError("No requirements found", s)
==> default: ValueError: ('No requirements found', '')
==> default: /vagrant/docker/splash.sh: line 17: $'\r': command not found
==> default: /vagrant/docker/splash.sh: line 18: $'\r': command not found
==> default: /vagrant/docker/splash.sh: line 19: $'\r': command not found
==> default: /vagrant/docker/splash.sh: line 20: $'\r': command not found
==> default: /vagrant/docker/splash.sh: line 21: $'\r': command not found
==> default: /vagrant/docker/splash.sh: line 22: $'\r': command not found
==> default: /vagrant/docker/splash.sh: line 23: $'\r': command not found
==> default: /vagrant/docker/splash.sh: line 24: $'\r': command not found
==> default: Installing portia dependencies
==> default: ==============================
==> default: /usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/package
s/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is
not available. This prevents urllib3 from configuring SSL appropriately and may
cause certain SSL connections to fail. For more information, see https://urllib3
.readthedocs.org/en/latest/security.html#insecureplatformwarning.
==> default:   InsecurePlatformWarning
==> default: /usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/package
s/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is
not available. This prevents urllib3 from configuring SSL appropriately and may
cause certain SSL connections to fail. For more information, see https://urllib3
.readthedocs.org/en/latest/security.html#insecureplatformwarning.
==> default:   InsecurePlatformWarning
==> default: /usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/package
s/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is
not available. This prevents urllib3 from configuring SSL appropriately and may
cause certain SSL connections to fail. For more information, see https://urllib3
.readthedocs.org/en/latest/security.html#insecureplatformwarning.
==> default:   InsecurePlatformWarning
==> default: /usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/package
s/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is
not available. This prevents urllib3 from configuring SSL appropriately and may
cause certain SSL connections to fail. For more information, see https://urllib3
.readthedocs.org/en/latest/security.html#insecureplatformwarning.
==> default:   InsecurePlatformWarning
==> default: Command "/usr/bin/python -c "import setuptools, tokenize;**file**='
/tmp/pip-build-Mzv4tB/Pillow/setup.py';exec(compile(getattr(tokenize, 'open', op
en)(**file**).read().replace('\r\n', '\n'), **file**, 'exec'))" install --record
 /tmp/pip-8fwDj1-record/install-record.txt --single-version-externally-managed -
-compile" failed with error code 1 in /tmp/pip-build-Mzv4tB/Pillow
==> default: /usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/package
s/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is
not available. This prevents urllib3 from configuring SSL appropriately and may
cause certain SSL connections to fail. For more information, see https://urllib3
.readthedocs.org/en/latest/security.html#insecureplatformwarning.
==> default:   InsecurePlatformWarning
==> default: Installing slyd as a Upstart service
==> default: ====================================
==> default: Starting slyd service
==> default: =====================
==> default: slyd start/running, process 15072

**Comment by ruairif** [comment]
Hi @tbarross I've rebuilt the vagrant VM on my own machine (debian based) and it worked without any issues (there are some red warnings during the build process but these are fine).
Could you try using `vagrant destroy` and then running `vagrant up` again and leave it run until it is finished. If it doesn't work for you again I'll try building it on windows and investigate what's happening.

**Comment by tbarross** [comment]
same thing, i installed docker which apparently updated my virtualbox to 5.0.6 (before i had 4. something)

i will give a shot on the docker and in last resource i will just build a debian VM and have Portia installed there.

but it would be great to know the issue (if its related to windows 8 or not)

**Comment by ruairif** [comment]
Can you tell me what version of vagrant you have too?

I'll build it on Windows this evening and I'll see what the problem is.

**Comment by tbarross** [comment]
ok let me just run the basic details in case i missed something before

win 8.1 64bit
virtualbox 5.0.6
vagrant 1.7.4
git => not sure what happen but when i just went to check its version i got a script error and couldnt get the "git" command on my windows cmd. currently installing 2.6.1 64bit and will retry to build everything from scratch.

**Comment by ruairif** [comment]
Hi @tbarross I managed to get it to work in Windows after some trial and error.
The reason it didn't built for me initially was because git was converting the `\n` line endings in the script file to `\r\n`.
To fix this I reinstalled git and told it to not convert line endings during checkout. I think the equivalent can be done using the following command:

    git config --global core.autocrlf input

After that I cloned the repo again ran `vagrant destroy` followed by `vagrant up` and everything worked as expected.
